packages/grid/x-grid/src/tests/selection.DataGridPro.test.tsx

```diff
@@ -70,9 +69,35 @@ describe('<DataGridPro /> - Selection', () => {
       );
       const selectAllCheckbox = document.querySelector('input[type="checkbox"]');
       fireEvent.click(selectAllCheckbox);
-      expect(getRow(0)).to.have.class('Mui-selected');
-      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
-      expect(getRow(1)).not.to.have.class('Mui-selected');
+      expect(Array.from(apiRef.current.getSelectedRows().keys())).to.deep.equal([0]);
+    });
+
+    it('should select all the lines when if checkboxSelectionVisibleOnly = false and pagination is not enabled', () => {
+      render(
+        <TestDataGridSelection
+          checkboxSelection
+          checkboxSelectionVisibleOnly={false}
+          rowLength={100}
+        />,
+      );
+
+      const selectAll = screen.getByRole('checkbox', {
+        name: /select all rows checkbox/i,
+      });
+      fireEvent.click(selectAll);
+      expect(apiRef.current.getSelectedRows().size).to.equal(100);
+    });
+
+    it('should select all the lines if checkboxSelectionVisibleOnly = true and pagination is not enabled', () => {
```

Comment:
This title doesn't reflect what the test is asserting.

```suggestion
    it('should throw a console error if checkboxSelectionVisibleOnly is used without pagination', () => {
```

Reply:
Oh I forgot to rename this one right